api/scpca_portal/models/library.py

```diff
@@ -72,3 +81,33 @@ def get_file_formats(sample_dir: Path):
             file_formats.append(Library.FileFormats.ANN_DATA)
 
         return file_formats
+
+    def get_data_file_paths(self) -> List[Path]:
+        """
+        Retrieves all data file paths on the aws input bucket associated with the Library object
+        and returns them as a list
+        """
+        project_id = self.samples.first().project.scpca_id
+        sample_id = (
+            self.samples.first().scpca_id
+            if self.samples.count() == 1
+            else ",".join([sample.scpca_id for sample in self.samples.all()])
+        )
```

Comment:
Grab this off of `self.metadata.get("sample_id")` I believe it will be available as a dict